### PR TITLE
Added if_exists check when drop the structures of index

### PR DESCRIPTION
**Pull request by wfystx**

### CODE REVIEW

File: src/meta/processors/indexMan/DropEdgeIndexProcessor.cpp [hunk 1]
```diff
@@ -18,7 +18,11 @@ void DropEdgeIndexProcessor::process(const cpp2::DropEdgeIndexReq& req) {
     auto edgeIndexID = getEdgeIndexID(spaceID, indexName);
     if (!edgeIndexID.ok()) {
         LOG(ERROR) << "Edge Index not exist Space: " << spaceID << " Index name: " << indexName;
-        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        if (req.get_if_exists()) {
```
Comment (Shylock-Hg) [inline src/meta/processors/indexMan/DropEdgeIndexProcessor.cpp]: Consider `?:` operator for more clear.

File: src/meta/processors/indexMan/DropTagIndexProcessor.cpp [hunk 2]
```diff
@@ -18,7 +18,11 @@ void DropTagIndexProcessor::process(const cpp2::DropTagIndexReq& req) {
     auto tagIndexID = getTagIndexID(spaceID, indexName);
     if (!tagIndexID.ok()) {
         LOG(ERROR) << "Tag Index not exist Space: " << spaceID << " Index name: " << indexName;
-        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        if (req.get_if_exists()) {
```
Comment (Shylock-Hg) [inline src/meta/processors/indexMan/DropTagIndexProcessor.cpp]: Ditto.

Comment (darionyaphet) [reply src/meta/processors/indexMan/DropTagIndexProcessor.cpp]: I think this is ok

File: src/meta/processors/indexMan/DropEdgeIndexProcessor.cpp [hunk 3]
```diff
@@ -18,11 +18,8 @@ void DropEdgeIndexProcessor::process(const cpp2::DropEdgeIndexReq& req) {
     auto edgeIndexID = getEdgeIndexID(spaceID, indexName);
     if (!edgeIndexID.ok()) {
         LOG(ERROR) << "Edge Index not exist Space: " << spaceID << " Index name: " << indexName;
-        if (req.get_if_exists()) {
-            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
-        } else {
-            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
-        }
+        resp_.set_code(
```
Comment (darionyaphet) [inline src/meta/processors/indexMan/DropEdgeIndexProcessor.cpp]: I think the original one is better

File: src/graph/test/IndexTest.cpp [hunk 4]
```diff
@@ -129,6 +129,18 @@ TEST_F(IndexTest, TagIndex) {
         code = client->execute(query, resp);
         ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
     }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "DROP TAG INDEX not_exists_tag_index";
+        auto code = client->execute(query, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
```
Comment (darionyaphet) [inline src/graph/test/IndexTest.cpp]: `ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code); `  would be better ?

**Pull request closed without merging**